src/main/java/com/cloudbees/jenkins/GitHubRepositoryName.java

```diff
@@ -223,12 +227,25 @@ public String toString() {
         return new NullSafeFunction<GitHub, GHRepository>() {
             @Override
             protected GHRepository applyNullSafe(@Nonnull GitHub gitHub) {
-                try {
-                    return gitHub.getRepository(format("%s/%s", repoName.getUserName(), repoName.getRepositoryName()));
-                } catch (IOException e) {
-                    LOGGER.warn("Failed to obtain repository {}", this, e);
-                    return null;
+                int mtries = 0;
+                while (mtries < MAX_RETRIES) {
```

Comment:
Is there some @annotation oriented retry mechanism in the project?
If so, can we use it?

Otherwise, can we look for some utility class/create one?